webapp/src/pages/registration/components/RegistrationComponent.js

```diff
@@ -0,0 +1,381 @@
+                    offer: result.form,
+                    error: result.error
+                }, () => {
+                    registrationService.getRegistrationForm(DEFAULT_EVENT_ID, this.state.offer.id).then(result => {
+                        if (result.error == "" && result.form.registration_sections.length > 0) {
+                            let questionSections = [];
+                            for (var i = 0; i < result.form.registration_sections.length; i++) {
+                                if (result.form.registration_sections[i].registration_questions.length > 0) {
+                                    questionSections.push(result.form.registration_sections[i]);
+                                }
+                            }
+                            registrationService.getRegistrationResponse().then(result => {
+                                if (result.error === "") {
+                                    this.setState({
+                                        isLoading: false,
+                                        answers: result.form.answers,
+                                        registrationId: result.form.registration_id
+                                    });
+                                }
+
+                            }).catch(error => { })
+                            this.setState({
+                                questionSections: questionSections,
+                                registrationFormId: result.form.id,
+                                isLoading: false
+                            });
+                        }
+                    });
+                });
+            }
+            else {
+                this.setState({
+                    isLoading: false
+                });
+            }
+        })
+    }
+
+    handleUploadFile = (file) => {
+        this.setState({
+            uploading: true
+        }, () => {
+            fileService.uploadFile(file, progressEvent => {
+                const percentCompleted = Math.round((progressEvent.loaded * 100) / progressEvent.total);
+                this.setState({
+                    uploadPercentComplete: percentCompleted
+                });
+            }).then(response => {
+                if (response.fileId && this.props.onChange) {
+                    this.props.onChange(this.props.question, response.fileId);
+                }
+                this.setState({
+                    uploaded: response.fileId !== "",
+                    uploadError: response.error,
+                    uploading: false
+                });
+            })
+        })
+    }
+
+    buttonSubmit = () => {
+
+        let data = {
+            registration_id: this.state.registrationId,
+            offer_id: this.state.offer.id,
+            registration_form_id: this.state.registrationFormId,
+            answers: this.state.answers
+        }
+        {
+            this.setState({
+                isLoading: true
```

Comment:
Rather set isSubmitting, isLoading will replace the whole page with a loading spinner.